### PR TITLE
feat: Add search view for places and worlds

### PR DESCRIPTION
**Pull request by jmoguilevsky**
This pr introduces the search for places and worlds in the overview, places and worlds pages, still missing in the favourites tab.

Also a fix in the sorting of the query of places was introduced to prevent duplicated results

### CODE REVIEW

File: src/components/Layout/NoResults.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  search: string
+}
+
+export default function NoResults(props: NoResultsProps) {
+  const { search } = props
+  const l = useFormatMessage()
+
+  return (
+    <>
+      <p className="search-results-header">
```
Comment (braianj) [inline src/components/Layout/NoResults.tsx:19]: can you use [BEM](https://getbem.com/naming/) format: like no-results__header

File: src/components/Layout/SearchInput.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import "./SearchInput.css"
+
+export default function SearchInput(
+  props: React.InputHTMLAttributes<HTMLInputElement>
+) {
+  return (
+    <input
+      {...props}
+      placeholder={props.placeholder || "Search..."}
+      className={TokenList.join(["SearchInput", props.className])}
```
Comment (braianj) [inline src/components/Layout/SearchInput.tsx:14]: can you use [BEM](https://getbem.com/naming/) format: like `search-input`

File: src/components/Layout/SearchList.css [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.search-results-header {
```
Comment (braianj) [inline src/components/Layout/SearchList.css:1]: can you use [BEM](https://getbem.com/naming/) format: like `search-results__header`

File: src/components/Layout/SearchList.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import "./SearchList.css"
+import "./OverviewList.css"
+
+export type SearchListProps = {
+  isLoadingPlaces: boolean
+  isLoadingWorlds: boolean
+  placeResultList?: AggregatePlaceAttributes[]
+  placeTotalResults?: number
+  worldResultList?: AggregatePlaceAttributes[]
+  worldTotalResults?: number
```
Comment (braianj) [inline src/components/Layout/SearchList.tsx:24]: should this be optionals? if one of the totals = undefined maybe `hasNoResults = false`?

File: src/components/Layout/SearchList.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    placeTotalResults === 0 &&
+    worldTotalResults === 0
+
+  if (hasNoResults) {
+    return <NoResults search={search} />
+  }
+
+  return (
+    <>
+      <p className="search-results-header">
```
Comment (braianj) [inline src/components/Layout/SearchList.tsx:61]: can you use [BEM](https://getbem.com/naming/) format: like `search-results__header`

File: src/hooks/useWorldListSearch.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    async () => {
+      if (search.length < 3) {
+        return { data: [], total: 0 }
+      }
+
+      const result = await Places.get().getWorlds({
+        ...options,
+        search,
+      })
+      return { ...result }
```
Comment (braianj) [inline src/hooks/useWorldListSearch.ts:23]: 🥲

File: src/pages/index.css [hunk 7]
```diff
@@ -70,3 +70,10 @@
     padding-right: 0;
   }
 }
+
+.search-results-header {
```
Comment (braianj) [inline src/pages/index.css:74]: can you use [BEM](https://getbem.com/naming/) format: like `search-results__header`

File: src/pages/index.tsx [hunk 8]
```diff
@@ -47,6 +57,17 @@ export default function OverviewPage() {
...
+    overviewOptions,
+    search
+  )
+  const [worldSearch, worldListSearchState] = useWorldListSearch(
+    overviewOptions,
+    search
+  )
+
+  const placeListSearch = placeSearch.data
+  const worldListSearch = worldSearch.data
```
Comment (braianj) [inline src/pages/index.tsx:72]: what do you think about sending this inside the `useMemo`

**Review by braianj — APPROVED** [review]
approving since fixes will come in a different PR

File: src/hooks/useWorldListSearch.ts (hunk 6 above)
Comment (jmoguilevsky) [reply src/hooks/useWorldListSearch.ts:23]: lol

**Pull request merged**